hadoop-tools/hadoop-azure/src/main/java/org/apache/hadoop/fs/azurebfs/AzureBlobFileSystemStore.java

```diff
@@ -966,29 +971,31 @@ AZURE_FOOTER_READ_BUFFER_SIZE, getAbfsConfiguration().getFooterReadBufferSize())
   public OutputStream openFileForWrite(final Path path,
       final FileSystem.Statistics statistics, final boolean overwrite,
       TracingContext tracingContext) throws IOException {
-    try (AbfsPerfInfo perfInfo = startTracking("openFileForWrite", "getPathStatus")) {
+    try (AbfsPerfInfo perfInfo = startTracking("openFileForWrite",
+        "getPathStatus")) {
       LOG.debug("openFileForWrite filesystem: {} path: {} overwrite: {}",
-              getClient().getFileSystem(),
-              path,
-              overwrite);
+          getClient().getFileSystem(),
+          path,
+          overwrite);
 
       String relativePath = getRelativePath(path);
       AbfsClient writeClient = getClientHandler().getIngressClient();
-
       final AbfsRestOperation op = getClient()
           .getPathStatus(relativePath, false, tracingContext, null);
       perfInfo.registerResult(op.getResult());
 
-      final String resourceType = getClient().checkIsDir(op.getResult()) ? DIRECTORY : FILE;
-      final Long contentLength = Long.valueOf(op.getResult().getResponseHeader(HttpHeaderConfigurations.CONTENT_LENGTH));
-
+      final String resourceType = getClient().checkIsDir(op.getResult())
+          ? DIRECTORY
+          : FILE;
       if (parseIsDirectory(resourceType)) {
         throw new AbfsRestOperationException(
-                AzureServiceErrorCode.PATH_NOT_FOUND.getStatusCode(),
-                AzureServiceErrorCode.PATH_NOT_FOUND.getErrorCode(),
-                "openFileForWrite must be used with files and not directories",
-                null);
+            AzureServiceErrorCode.PATH_NOT_FOUND.getStatusCode(),
+            AzureServiceErrorCode.PATH_NOT_FOUND.getErrorCode(),
+            "openFileForWrite must be used with files and not directories",
+            null);
       }
+      final Long contentLength = Long.valueOf(op.getResult()
```

Comment:
We can use the extractContentLength() method proposed above here as well.

Reply:
Taken